### PR TITLE
[FEATURE] Add support for returning indices smaller than `disk_space` in space filter

### PR DESCRIPTION
**Pull request by dustin-decker**
Addresses https://github.com/elastic/curator/issues/1066

I'm not attached to the `index_comparison` key at all, and hope there's a better name for it. I'm open to suggestions.
I'm not very familiar with asciidoc, so I hope I did that part right. If you want me to make any changes, just let me know.

Thanks a lot for Curator. It's a great tool, and I make heavy use of it.

### CODE REVIEW

**Comment by untergeek** [comment]
Let me think about the name of the flag.  I like what you've done here, especially guaranteeing reverse-compatibility.  But this is a big shift in what the filter does.  I definitely want the name to be a good fit, and `index_comparison` doesn't quite seem the right fit.  I will think about this for a bit.

**Comment by sevoma** [comment]
Maybe a bool? `smaller_than=True` (default False)

**Comment by untergeek** [comment]
While I still consider the naming for this, will you please rebase against the master branch?  This commit has conflicts now.

**Comment by dustin-decker** [comment]
Rebased

**Comment by untergeek** [comment]
Thanks for rebasing.  

I like the idea of preserving the current behavior with a named approach, `threshold_behavior` as the key, with acceptable values of `less_than` and `greater_than`.  Of course, `greater_than` would be the default behavior to preserve reverse compatibility.

Thoughts?

**Comment by dustin-decker** [comment]
I think that's a big improvement in clarity. What do you think of `disk_space_behavior` for the key? You already have the `disk_space` key, so it's slightly more obvious than 'threshold'. It also places it in close proximity to disk_space when alphabetically sorted, which is helpful for documentation.
I'll try to get the changes done, whatever the end up being, tomorrow or the day after.

**Comment by untergeek** [comment]
I'm not a fan of mixing `disk_space` and `behavior` because the disk space isn't _behaving._  We have a threshold that is established by the value of `disk_space`, and then we need to do something with that value.  Here are some other possibilities:

* `threshold_calculation`
* `threshold_direction`
* `threshold_measurement`
* `threshold_comparison`

File: curator/defaults/filter_elements.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ def timestring(**kwargs):
     else:
         return { Optional('timestring', default=None): Any(str, unicode, None) }
 
+def threshold_behavior(**kwargs):
+    # This setting is only used with the space filtertype and defaults to 'greater_than'.
+    return { Optional('threshold_behavior', default='larger'): Any('greater_than', 'less_than') }
```
Comment (untergeek) [inline curator/defaults/filter_elements.py]: This is going to cause a problem:

```
default='larger'
```

This has to be `greater_than` or we'll have serious issues.

Comment (dustin-decker) [reply curator/defaults/filter_elements.py]: Good catch. I'm still reviewing my changes :)

Comment (dustin-decker) [reply curator/defaults/filter_elements.py]: I should probably have a test for the default behavior

**Pull request merged**